base_layer/core/src/chain_storage/db_transaction.rs

```diff
@@ -289,7 +284,8 @@ pub enum WriteOperation {
         proof_hash: HashOutput,
         mmr_position: u32,
     },
-    Delete(DbKey),
+    DeleteHeader(u64),
```

Comment:
Yeah, I looked at it and would prefer to use the hash, but the reorgs don't read out the headers at the moment. This would be a slightly more involved implementation and would involve a read at each step. I think there may be a way of doing it, but I'd have to look deeper into it. 

Out of interest, I originally put in the hash and went a fair way down the line implementing before reverting it. I may change it to use a hash in a followup pr